src/main_functions.c

```diff
@@ -0,0 +1,530 @@
+			*quit = 1;
+		} else if (!strcmp(line, "c")) {
+            
+            while ((args == NULL) ^ i) {    // either user returns to main page, or inputs a vaid new destination
+                printf("\n\nEnter a viable file name (*.txt), or enter 'r' to return to the save page.\n\n");
+                shell_prompt();
+                verify = scanf("%s", line);
+                assert (!(verify < 0));
+
+                args = strstr(line, ".txt");
+                if (!strcmp(line, "r")) i = 1;
+            }
+
+            *quit = 1;
+            if (i == 0) save_corrections(line, lines);  // save to different file destination
+
+		} else if (!strcmp(line, "r")) {
+			*quit = 0;
+		} else if (!strcmp(line, "q")) {
+			*quit = 1;
+		} else {
+			shell_error("Please type in one of the indicated commands!");   // wrong input
+			i = 1;
+		}
+
+	}
+}
+
+/*
+ *	II. Functions for editing strings
+ */
+
+void underline_misspelled(char *word, char* underline)
+{
+	int j = strlen(word);
+    int i = 0;
+	for(; i < j; i++) {
+    	strcat(underline, "^");
+	}
+		strcat(underline, " ");
+		// printf("start \n");
+		// printf("%s", underline);
+		// printf("end \n");
+
+}
+
+void underline_correct_spelling(char *word, char* underline)
+{
+	int j = strlen(word);
+    int i = 0;
+	for(; i < j; i++) {
+    		strcat(underline, " ");
+	}
+		strcat(underline, " ");
+
+}
+
+char* underline_misspelled_sentence(char** misspelled, char* sentence, int element) {
+
+	char* underline = malloc(strlen(sentence));
+	underline[0] = '\0'; 
+
+	while(misspelled[element] != NULL) {
+
+		char *ptr = strstr(sentence, misspelled[element]);
+
+		if(ptr != NULL) {
+
+			int pos = ptr - sentence;
+
+			int i = 0;
+			size_t j = 0;
+
+			for(; i < pos; i++) {
+
+				strcat(underline, " ");
+
+			}
+
+			for (; j < strlen(misspelled[element]); j++) {
+
+				strcat(underline, "^");
+
+			}
+
+			sentence = ptr+strlen(misspelled[element]);
+
+		}
+
+		element++;
+
+	}
+
+	return underline;
+
+}
+
+
+
+int add_to_misspelled(char *word, char** misspelled)
+{
+	if (word == NULL || misspelled == NULL) {
+		return EXIT_FAILURE;
+	}
+	int i = 0;
+	while(misspelled[i] != NULL) {
+		i++;
+	}
+	misspelled[i] = word;
+	return EXIT_SUCCESS;
+	// printf("adding badword %s", badword);
+	// printf("i is %d ", i);
+}
+
+int parse_string(char* string, dict_t *dict, char *underline, char** misspelled)
+{
+	char *tkn = strtok(string, ": ,.-'\n'" ""); //words only separated by these punctuation
+	while (tkn != NULL) {
+
+		if (valid_word(tkn, dict) == EXIT_FAILURE){
+			//printf("%s ", tkn);
+			underline_misspelled(tkn, underline);
+			add_to_misspelled(tkn, misspelled);
+		}
+		else if (valid_word(tkn, dict) == EXIT_SUCCESS) {
+			//printf("%s ", tkn);
+			underline_correct_spelling(tkn, underline);
+		}
+		else {
+			printf("error processing text");
+			return EXIT_FAILURE;
+		}
+		tkn = strtok(NULL, " ,.-");
```

Comment:
What is this line for?

Reply:
@sarikam1

Reply:
Think it's necessary to move on to the next word and parse it